src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from 'react';
+import classNames from 'classnames';
+import { FILTERS } from '../../constants/filters';
+
+interface Props {
+  onSetActiveFilter: React.Dispatch<React.SetStateAction<FILTERS>>;
+}
+
+export const Filter: React.FC<Props> = ({ onSetActiveFilter }) => {
+  const [selectedFilter, setSelectedFilter] = useState(FILTERS.ALL);
+
+  const handleFilter = (filterType: FILTERS) => {
+    onSetActiveFilter(filterType);
+    setSelectedFilter(filterType);
```

Comment:
you doing here pretty same thing, so you can pass `selectedFilter` as props to this component and avoid using extra state